src/libraries/System.Private.CoreLib/src/System/IO/Path.cs

```diff
@@ -926,5 +926,69 @@ private static string GetRelativePath(string relativeTo, string path, StringComp
         /// Returns true if the path ends in a directory separator.
         /// </summary>
         public static bool EndsInDirectorySeparator(string path) => PathInternal.EndsInDirectorySeparator(path);
+
+        /// <summary>
+        /// Removes any redundant segments found in the specified path.
+        /// </summary>
+        /// <param name="path">A string containing an absolute or relative path that may or may not contain redundant segments.</param>
+        /// <returns><paramref name="path" /> without any redundant segments.</returns>
+        [return: NotNullIfNotNull("path")]
+        public static string? RemoveRedundantSegments(string? path)
```

Comment:
Any reason why the argument and the returned value are nullable?
This does not match the [accepted](https://github.com/dotnet/runtime/issues/2162#issuecomment-534695882) API shape.

Reply:
It's my understanding that certain elements like nullability markers or the unsafe keyword are independent from the final approved API shape.

Reply:
@terrajobst Can you confirm this? Why would we want to accept a null *path*?

Reply:
You can read this comment thread for a confirmation from both @jkotas and @JeremyKuhne:

https://github.com/dotnet/runtime/pull/2187#discussion_r381461941